### DISCUSSION TITLE
How to listen on events or any messages that get to sent

### DISCUSSION THREAD

**Discussion by tonislcs** [supabase-community/supabase-csharp#130] [General]
So am trying to listen on any events or specific events in realtime channel how would i go ahead and do that?

for example it ried doing something like this
`            var channel = supabase.Realtime.Channel("1");
            await channel.Subscribe();
            channel.AddMessageReceivedHandler((sender, args) =>
            {
                Console.WriteLine(args.Payload);
            });` in Supabase Viewer i could see that there was stuff being sent but everytime it was sent in was just a empty `Supbase.Realtime.Socket.SocketResponsePayload`

**Comment by acupofjose** [comment]
Try checking here!

https://github.com/supabase-community/realtime-csharp#broadcast

**Comment by tonislcs** [comment]
Thanks what's the difference between supabase-csharp package and realtime-csharp does it just have realtime part of supabase-csharp?

**Reply by acupofjose** [reply]
Correct, it’s just the repo that contains all the realtime functionality.

**Reply by tonislcs** [reply]
Is there a way to listen on a specific event name?

**Reply by acupofjose** [reply]
Personally, I'd just use a different channel per event type you're looking for. The channel names are totally arbitrary. But otherwise, not currently, you'd have to filter them by event name yourself!
